butane_core/src/codegen/dbobj.rs

```diff
@@ -29,12 +29,43 @@ pub fn impl_dbobject(ast_struct: &ItemStruct, config: &Config) -> TokenStream2 {
     let pklit = make_ident_literal_str(&pkident);
     let auto_pk = is_auto(&pk_field);
 
+    let values: Vec<TokenStream2> = push_values(ast_struct, |_| true);
     let insert_cols = columns(ast_struct, |f| !is_auto(f));
-    let save_cols = columns(ast_struct, |f| !is_auto(f) && f != &pk_field);
 
-    let mut post_insert: Vec<TokenStream2> = Vec::new();
-    add_post_insert_for_auto(&pk_field, &mut post_insert);
-    post_insert.push(quote!(self.state.saved = true;));
+    let save_core = if is_auto(&pk_field) {
+        let pkident = pk_field.ident.clone().unwrap();
+        let values_no_pk: Vec<TokenStream2> = push_values(ast_struct, |f: &Field| f != &pk_field);
+        let save_cols = columns(ast_struct, |f| !is_auto(f) && f != &pk_field);
+        quote!(
+            // Since we expect our pk field to be invalid and to be created by the insert,
+            // we do a pure insert or update, no upsert allowed. Note that some database backends
```

Comment:
Why cant we use upset for updates?  It seems like we could.

Reply:
Well we could, but at that point we know the object has already been created so there's no real point. We're in a separate `quote!` block from the the upsert so I don't think we can merge easily. Let me know if you see something more elegant though.
I'll reword slightly to try to be more clear.

Reply:
Ya, it isnt beneficial to use upsert. "no upsert allowed" seemed a bit too strongly asserting it would be a problem.
Rewording the comment is sufficient.